### PR TITLE
Derive direction from document element too

### LINKED ISSUE #327: Add dir/lang member on dictionaries whose content is displayed in browser UI
As we can't mark-up the labels in dictionaries (using HTML), we might need to add `dir` ("ltr" and "rtl") members on dictionaries that contain things that are displayed to the end-user (defaulting to "ltr" when missing). So:

```JS
[
  displayItems: [{
    label: "البند الخاص (للبيع!)",
    dir: "rtl",
    lang: "ar-AE",
    amount: { },
  }]
]
```

Have the i18n folks looked at this API? They will likely ask us to add that. We might also need `lang`. 

cc @r12a.

### PR DESCRIPTION
**Pull request by marcoscaceres**
Related to, but does not close #327, #946, #948, #951

The following tasks have been completed:

 * [ ] Confirmed there are no ReSpec errors/warnings.
 * [ ] Modified Web platform tests (link)
 * [ ] Modified MDN Docs (link)
 * [ ] Has undergone security/privacy review (link)
 
Implementation commitment:

 * [ ] Safari (link to issue)
 * [ ] Chrome (link to issue)
 * [ ] Firefox (link to issue)
 * [ ] Edge (public signal)

Optional, impact on Payment Handler spec?

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/payment-request/pull/956.html" title="Last updated on May 20, 2021, 2:40 AM UTC (d894b79)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/payment-request/956/79e23d4...d894b79.html" title="Last updated on May 20, 2021, 2:40 AM UTC (d894b79)">Diff</a>

### CODE REVIEW

**Comment by ianbjacobs** [comment]
Hi @marcoscaceres,

Thanks for starting this. Some thoughts:

1) The phrase "payment sheet" is not used elsewhere in the document. I think we should either define it (e.g., by adding "(called the 'payment sheet')" after "Present a user interface" at the beginning of item 18) or remove it. If we remove it, then the note title could become: "Localization of the user interface to interact with handlers". Then you could just s/payment sheet/user interface/ in the note body.

2) Point 18 says "SHOULD use info provided in content." It's not clear to me that the note needs to say "As such, user agents will generally present the payment sheet using the user agent's default language." That to me suggests that despite the recommendation we just made, nobody is doing it.

What about a Note that says:

 "This API relies on inherited localization information. The working group is considering support for fine-grained localization of the user interface and of individual PaymentItems in a future version of this API."

**Review by ianbjacobs — CHANGES_REQUESTED** [review]
(See my comments above)

**Comment by marcoscaceres** [comment]
> That to me suggests that despite the recommendation we just made, nobody is doing it.

That's the truth tho :) 

Will update based on your suggestions.

**Review by ianbjacobs — APPROVED** [review]

**Comment by ianbjacobs** [comment]
@aphillips, would you mind having a look at this proposal to address the lack of fine-grain localization in V1 of the API? Thank you!

File: index.html [hunk 1]
```diff
@@ -1139,20 +1139,23 @@ <h2>
...
               with the |handlers|. The user agent SHOULD prioritize the user's
               preference when presenting payment methods. The user interface
-              SHOULD be presented using the language and locale-based
-              formatting that matches the |document|'s [=document
-              element|document element's=] [=Node/language=], if any, or an
-              appropriate fallback locale if that is not available.
+              SHOULD be presented using the language, base direction, and
+              locale-based formatting that matches the |document|'s [=document
+              element|document element's=] [=Node/language=] and
+              {{Document/dir}} attribute, if any, or an appropriate fallback
```
Comment (aphillips) [inline index.html]: The direction should also be at the Node level, no?

File: index.html [hunk 2]
```diff
@@ -1139,20 +1139,23 @@ <h2>
...
+              locale if that is not available.
```
Comment (aphillips) [inline index.html]: Omit `locale` since this is both locale and direction now?

File: index.html [hunk 3]
```diff
@@ -1139,20 +1139,23 @@ <h2>
...
-                using the user agent's default language. The working group is
-                contemplating adding the ability for developers to specify the
-                language of the payment sheet, and of specific
-                {{PaymentItems}}, in a future version of this API.
+                The API does not currently provide a way for developers to
+                specify the language and base direction in which the payment
+                sheet is presented to end users. Instead, the API relies on
+                localization information inherited from the document. The
+                working group is considering support for fine-grained
+                localization of the user interface and of individual
```
Comment (aphillips) [inline index.html]: I'd probably change "fine-grained" to "control of" (too strong?) or perhaps "requesting  specific"

**Review by aphillips — CHANGES_REQUESTED** [review]
Generally looks good. Small tweaks proposed.

File: index.html (hunk 1 above)
Comment (marcoscaceres) [reply index.html]: It should, but there is no such thing on Node unfortunately:
https://dom.spec.whatwg.org/#node
 
The only "dir"s in the platform are: 
https://respec.org/xref/?term=dir

Comment (marcoscaceres) [reply index.html]: Oh, wait... we might be able to use https://html.spec.whatwg.org/multipage/dom.html#the-directionality ... but I'll need to get it "exported" so we can use it in our spec.

Comment (marcoscaceres) [reply index.html]: Sent https://github.com/whatwg/html/pull/6676 ...

Comment (aphillips) [reply index.html]: @marcoscaceres Thanks. I squinted at `{{HTMLElement/dir}}` and it wasn't clear to me if it included inherited direction. `[^html-global/dir^]` is the definition of the `dir` attribute and includes inheritance, but doesn't seem quite to match our needs here. Exporting `directionality` is a better solution--not just here but generally! I may take whatwg/html#6676 to expedite ;-).

Comment (aphillips) [reply index.html]: LOL. I thought 6676 was an issue, but it's a PR...

Comment (marcoscaceres) [reply index.html]: @aphillips, take a look at @annevk's comments at https://github.com/whatwg/html/pull/6676 ... I think they make a lot of sense. 

My feeling is that we punt on the direction for now, and solve this with "localizable" in a future version of the spec.

Comment (ianbjacobs) [reply index.html]: @aphillips, would that work for you?

Comment (aphillips) [reply index.html]: LGTM.

@marcoscaceres I'm pondering a reply to @annevk: I think he is right, although I18N does want to promote using the document's context when it is available and is appropriate.

**Review by aphillips — APPROVED** [review]

**Pull request merged**